plugins/modules/proxmox_snap_info.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020, Jeffrey van Pelt (@Thulium-Drake) <jeff@vanpelt.one>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_snap_info
+short_description: Snapshot management of instances in Proxmox VE cluster
+version_added: 6.1.0
+description:
+  - Allows you to create/delete/restore snapshots from instances in Proxmox VE cluster.
+  - Supports both KVM and LXC, OpenVZ has not been tested, as it is no longer supported on Proxmox VE.
+options:
+  hostname:
+    description:
+      - The instance name.
+    type: str
+  vmid:
+    description:
+      - The instance id.
+      - If not set, will be fetched from PromoxAPI based on the hostname.
+    type: str
+  timeout:
+    description:
+      - Timeout for operations.
+    default: 30
+    type: int
+  snapname:
+    description:
+      - Beginning of the snapshot name getsnapshots takes care of.
+    default: ''
```

Comment:
How about not having a default, and testing whether the option's value is `not None`?